exercises/exercise_3/README.md

```diff
@@ -1,266 +1,255 @@
+    Removing service voting-app_result
+    Removing service voting-app_vote
+    Removing service voting-app_worker
+    Removing network back-tier
+    Removing network front-tier
+    Application "voting-app" uninstalled on context "swarm"
+    ```
+    </details>
+
+    :tada: It's gone now!
+
+3. To verify, you can run `docker app ls --target-context swarm` and validate that it's gone.
+
+    <details>
+      <summary>Full output</summary>
+    
+    ```console
+    $ docker app ls --target-context swarm
+    INSTALLATION APPLICATION LAST ACTION RESULT CREATED MODIFIED REFERENCE
+    ```
+    </details>
+
+
+## Pushing our Docker App
+
+Now that we've learned how to deploy a locally-sourced application, let's push our app to Docker Hub and deploy from there!
+
+1. Let's first take a look at the various options and flags on the `docker app push` command.
+
+    <details>
+      <summary>Full output</summary>
+    
+    ```console
+    $ docker app push --help
+
+    Usage:  docker app push [APP_NAME] --tag TARGET_REFERENCE [OPTIONS]
+
+    Push an application package to a registry
+
+    Examples:
+    $ docker app push myapp --tag myrepo/myapp:mytag
+
+    Options:
+          --insecure-registries strings   Use HTTP instead of HTTPS when pulling from/pushing to those registries
+          --platform strings              For multi-arch service images, only push the specified platforms
+      -t, --tag string                    Target registry reference (default: <name>:<version> from metadata)
+    ```
+    </details>
+
+2. When pushing, we will tag the application to include your Docker Hub account (`<your-hub-username>/voting-app.dockerapp`) and specify the tag as the current version (which defaulted to `0.1.0`). While not required, we do encourage you to keep the `.dockerapp` suffix to the image name to help make it clear that that repo _is_ a Docker App bundle.
+
+    <details>
+      <summary>Solution/Full Output</summary>
+    
+    ```console
+    $ docker app push --tag mikesir87/voting-app.dockerapp:0.1.0
+    docker.io/mikesir87/voting-app.dockerapp:0.1.0-invoc
+    mikesir87/examplevotingapp_vote
+    sha256:a0d4d29d...: Skip (already present)
+    redis:alpine
+    sha256:ef67270b...: Skip (already present)
+    postgres:9.4
+    sha256:094e3a9e...: Skip (already present)
+    dockersamples/examplevotingapp_worker
+    sha256:55753a7b...: Skip (already present)
+    mikesir87/examplevotingapp_result
+    sha256:69198c25...: Skip (already present)
+    WARN[0003] reference for unknown type: application/vnd.cnab.config.v1+json
+    Successfully pushed bundle to docker.io/mikesir87/voting-app.dockerapp:0.1.0. Digest is sha256:ca706ede7e387173cf28b20e65336299873c072deb422c35a0fd57379b46932e.
+    ```
+    </details>
+
+    The output may look slightly different, as we had already previously pushed the app before capturing the output to display above. However, you should see a "Successfully pushed bundle" message at the end.
```

Comment:
Make them notice we also output the digest, like for a `docker push` ? And that the app is also content addressable, like any image ?